### PR TITLE
feat(use-mounted-state): add useMountedState react hook

### PR DESCRIPTION
**Pull request by andresin87**
**TASK**: 

### Types of changes

- [x] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/hooks/use-mounted-state/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "@spark-ui/use-mounted-state",
+  "version": "0.0.1",
+  "description": "",
```
[the remark this answers is not in the document]
Comment (andresin87) [reply packages/hooks/use-mounted-state/package.json:4]:
```suggestion
  "description": "check the component's mount state",
```

File: packages/hooks/use-mounted-state/src/useMountedState.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+export function useMountedState(): () => boolean {
+  const mountedRef = useRef<boolean>(false)
```
Comment (acd02) [inline packages/hooks/use-mounted-state/src/useMountedState.tsx:4]:
```suggestion
  const mountedRef = useRef(false)
```

File: packages/hooks/use-mounted-state/src/useMountedState.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  const get = useCallback(() => mountedRef.current, [])
```
Comment (acd02) [inline packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: Question, why return a function rather than a boolean directly?

Comment (andresin87) [reply packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: yes, the intention is to use it in any place calling directly where you need, not infering the state calculated in the resolved hook placement

Comment (andresin87) [reply packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: this is taken from react-use library

Comment (acd02) [reply packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: 👍 , but do we need to re-rewrite it and expose it then? Couldn't we use the one from the react-use lib instead?

Comment (andresin87) [reply packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: this are kind of primitives, we should provide and maintain set of common tools  for repetitive behaviors in components.

Comment (andresin87) [reply packages/hooks/use-mounted-state/src/useMountedState.tsx:6]: this is the simplest one, its just to add the useCombinedState, which uses it.

**Pull request merged**